src/main/java/com/flightstats/hub/webhook/strategy/TimedWebhookStrategy.java

```diff
@@ -157,15 +163,9 @@ public ContentPath getLastCompleted() {
         return clusterCacheDao.getOrNull(webhook.getName(), WEBHOOK_LAST_COMPLETED);
     }
 
-    @Override
-    public void start(Webhook webhook, ContentPath startingPath) {
-        ThreadFactory factory = new ThreadFactoryBuilder().setNameFormat(webhook.getBatch() + "-webhook-" + webhook.getName() + "-%s").build();
-        executorService = Executors.newSingleThreadScheduledExecutor(factory);
-        log.info("starting {} with starting path {}", webhook, startingPath);
-        executorService.scheduleAtFixedRate(new Runnable() {
-
-            ContentPath lastAdded = startingPath;
-
+    @VisibleForTesting
+    Runnable getContentKeyGenerator(Webhook webhook, ContentPath initialStartingPath) {
```

Comment:
This was extracted from the start method so it could be tested without messing with threads.